### PR TITLE
Admin merchant index #22

### LINKED ISSUE #22: User Story 48, Admin visits Merchant Index Page
```
As an admin user
When I visit the merchant's index page at "/merchants"
I see all merchants in the system
Next to each merchant's name I see their city and state
The merchant's name is a link to their Merchant Dashboard at routes such as "/admin/merchants/5"
I see a "disable" button next to any merchants who are not yet disabled
I see an "enable" button next to any merchants whose accounts are disabled
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #25

child of #79

depends on #24

### PR DESCRIPTION
**Pull request by allieIsOkay**
-Adds tests for admin merchant index page
-Adds logic for presentation of admin privileged content to merchant index page
- Removes User::enabled_merchants test and functionality following necessity of all users being present on page for admin viewing.
- Adds User::all_merchants tests
- Adds User::all_merchants functionality
- moves current_admin method from Admin::BaseController to ApplicationController and adds helper method to aid in admin specific presentation logic when viewing merchant index
- Adds namespaced merchants show route to routes in order to have routing for /admin/merchants/id per spec for admin access to merchant dashboard. Controller selection pending further group discussion.
closes #22

### CODE REVIEW

**Comment by csvlewis** [comment]
Good work! There's probably places where the all_merchants method could be useful in other parts of the project.

**Review by csvlewis — APPROVED** [review]

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -15,4 +15,8 @@ def current_user
   def current_shopper?
     !(current_user && (current_user.admin? || current_user.merchant?))
   end
+  
+  def current_admin?
```
Comment (teresa-m-knowles) [inline app/controllers/application_controller.rb:19]: @plapicola why did we move this here from the base controller? just trying to understand!

Comment (allieIsOkay) [reply app/controllers/application_controller.rb:19]: By putting it in the application controller and making it a helper method, we're able to use it in the views, including ones outside the admin namespace where it existed before. For this case specifically I did it so that I can control if all merchants(including disabled merchants) are displayed on the merchant index based on the current user's role. Hope this helps!

**Review by timnallen — APPROVED** [review]
This looks great! Nice job!
